File: model_zoo/vision/ultraface/cpp/ultraface.cc
// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#include "fastdeploy/vision.h"

int main() {
  namespace vis = fastdeploy::vision;

  auto model = vis::linzaer::UltraFace("version-RFB-320.onnx");
  if (!model.Initialized()) {
    std::cerr << "Init Failed! Model: " << model_file << std::endl;
    return -1;
  } else {
    std::cout << "Init Done! Model:" << model_file << std::endl;
  }
  model.EnableDebug();

  cv::Mat im = cv::imread("3.jpg");
  cv::Mat vis_im = im.clone();

  vis::FaceDetectionResult res;
  if (!model.Predict(&im, &res, 0.7f, 0.3f)) {
    std::cerr << "Prediction Failed." << std::endl;
    return -1;
  } else {
    std::cout << "Prediction Done!" << std::endl;
  }

  // 输出预测框结果
  std::cout << res.Str() << std::endl;

  // 可视化预测结果
  vis::Visualize::VisFaceDetection(&vis_im, res, 2, 0.3f);
  cv::imwrite("vis_result.jpg", vis_im);
  std::cout << "Detect Done! Saved: " << vis_path << std::endl;
  return 0;
}

File: model_zoo/vision/ultraface/ultraface.py
import fastdeploy as fd
import cv2

# 下载模型
model_url = "https://github.com/Linzaer/Ultra-Light-Fast-Generic-Face-Detector-1MB/raw/master/models/onnx/version-RFB-320.onnx"
test_img_url = "https://github.com/Linzaer/Ultra-Light-Fast-Generic-Face-Detector-1MB/raw/master/imgs/3.jpg"
fd.download(model_url, ".", show_progress=True)
fd.download(test_img_url, ".", show_progress=True)

# 加载模型
model = fd.vision.linzaer.UltraFace("version-RFB-320.onnx")

# 预测图片
im = cv2.imread("3.jpg")
result = model.predict(im, conf_threshold=0.7, nms_iou_threshold=0.3)

# 可视化结果
fd.vision.visualize.vis_face_detection(im, result)
cv2.imwrite("vis_result.jpg", im)

# 输出预测结果
print(result)
print(model.runtime_option)
